packages/components/src/Form/Form.tsx

```diff
@@ -86,11 +87,8 @@ export const Form = forwardRef(function InternalForm(
   }
 
   function errorHandler(errs: FieldErrors<FieldValues>) {
-    const firstErrName = Object.keys(errs)[0];
-    const element = document.querySelector(
-      `[name="${firstErrName}"]`,
-    ) as HTMLElement;
+    const errorKeys = Object.keys(errs);
 
-    element?.focus();
+    setFocus(errorKeys[0]);
   }
 });
```

Comment:
Any reason we couldn't leverage the built-in functionality from RHF? https://react-hook-form.com/docs/useform#shouldFocusError

Reply:
I was trying that and it didn't seem to work as expected. I can try again and see if it was a weird caching thing

Reply:
Wonder if it's an issue with the register note 🤔

Reply:
It turns out that after I added the refs in FormField it worked as expected. I assume it had to do with 
```
Only registered fields with a ref will work. 
```

and a ref wasn't registered until my change

Reply:
I haven't checked the code, but looks like the use controller ref would need to be passed to the component? https://react-hook-form.com/docs/usecontroller

Reply:
Nice!